dev/tools/gen_keycodes/data/keyboard_key.tmpl

```diff
@@ -165,18 +196,36 @@ class LogicalKeyboardKey extends KeyboardKey {
   /// “o” for [keyLabel], but would return “ö” for [RawKeyEvent.character].
   ///
   /// {@macro flutter.services.RawKeyEventData.keyLabel}
-  final String keyLabel;
-
-  @override
-  int get hashCode => keyId.hashCode;
+  String get keyLabel {
+    return _unicodeKeyLabel(keyId)
+        ?? _keyLabels[keyId]
+        ?? '';
+  }
 
-  @override
-  bool operator ==(Object other) {
-    if (other.runtimeType != runtimeType) {
-      return false;
-    }
-    return other is LogicalKeyboardKey
-        && other.keyId == keyId;
+  /// The debug string to print for this keyboard key, which will be null in
+  /// release mode.
+  ///
+  /// This is different from [keyLabel] for 2 cases:
+  ///
+  ///  * When the key is a printable key, it returns a more formal name
```

Comment:
It's not just when it's printable: it will return a string for things like "Volume Up" and the (oh so commonly used!) "Launch Word Processor" key, etc. too.

Reply:
I rewrote the 2 docs and listed how they're defined separately.